src/main/java/ca/uhn/fhir/jpa/starter/Interceptors/RemovePatientExtensionsInterceptor.java

```diff
@@ -17,23 +20,42 @@
 @Interceptor
 public class RemovePatientExtensionsInterceptor {
 
+	private static final Logger log = LoggerFactory.getLogger(RemovePatientExtensionsInterceptor.class);
+
+	/**
+	 * This function hooks into the server outgoing responses and removes
+	 * any pseudonym extensions that patients might have.
+	 *
+	 * @param resource
+	 * 	The resource of the outgoing response
+	 */
 	@Hook(Pointcut.SERVER_OUTGOING_RESPONSE)
-	public void removeExtensions(IBaseResource resource) {
+	public void checkForPatient(IBaseResource resource) {
+		log.info("Removing patient pseudonym extensions");
 		if (resource instanceof Patient patient) {
-			if (patient.hasExtension()) {
-				patient.setExtension(Collections.emptyList());
-			}
+			removeExtension(patient);
 		}
 		if (resource instanceof Bundle bundle) {
```

Comment:
I know this is not your code. But I believe in leaving code better than you found it. So, I encourage people whenever they can, to do the same. By the way, you don't have to if you don't wish to, it's just a recommendation.

Try to reduce deeply nested loops